api/src/object/ShopScrape.ts

```diff
@@ -150,6 +156,23 @@ export class ShopScrape implements DurableObject {
         }
     }
 
+    async shouldNotify(user: User, notificationKey: string): Promise<boolean> {
```

Comment:
Can you use the KV for this? Set a item a expire and just check the existence. Cloudflare will clean up the key. it's kind of like redis
https://developers.cloudflare.com/kv/api/write-key-value-pairs/#create-expiring-keys

In that way, we don't have to talk with the SQL server, and we control it fully, the user can delete the notification.